spec/models/manageiq/providers/embedded_ansible/automation_manager/playbook_runner_spec.rb

```diff
@@ -140,6 +138,41 @@
     end
   end
 
+  describe '#post_ansible_run' do
+    let(:options) { {:inventory => 'inv1', :job_template_ref => 'jt1'} }
+
+    context 'playbook runs successfully' do
+      it 'removes temporary inventory and job template and finishes the job' do
+        expect(subject).to receive(:delete_inventory)
+        expect(subject).to receive(:delete_job_template)
+        subject.post_ansible_run('Playbook ran successfully', 'ok')
+        expect(subject.state).to eq('finished')
+        expect(subject.status).to eq('ok')
+      end
+    end
+
+    context 'playbook runs with error' do
+      it 'removes temporary inventory and job template and finishes the job with error' do
+        expect(subject).to receive(:delete_inventory)
+        expect(subject).to receive(:delete_job_template)
+        subject.post_ansible_run('Ansible engine returned an error for the job', 'error')
+        expect(subject.state).to eq('finished')
+        expect(subject.status).to eq('error')
+      end
+    end
+
+    context 'cleaning up has error' do
+      it 'does fail the job but logs the error' do
+        allow(subject).to receive(:delete_inventory)
+        allow(subject).to receive(:temp_configuration_script).and_raise('fake error')
+        expect($log).to receive(:log_backtrace)
+        subject.post_ansible_run('Playbook ran successfully', 'ok')
```

Comment:
In the case that a `job_template` or `inventory` record is not removed successfully - I agree that an error shouldn't be raised, but it seems inconsistent to leave the message 'Playbook ran successfully.'  Possibly it should be modified to say 'Playbook ran successfully with cleanup errors' or something along those lines.

Reply:
I now appended cleanup errors to the existing message.